pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/inv/json/BaseJsonIndexCreator.java

```diff
@@ -130,7 +132,7 @@ void addToPostingList(String value) {
   void generateIndexFile()
       throws IOException {
     ByteBuffer headerBuffer = ByteBuffer.allocate(HEADER_LENGTH);
-    headerBuffer.putInt(VERSION);
+    headerBuffer.putInt(VERSION_2);
```

Comment:
why do we need to bump the version when the index generation logic does not change?

Reply:
The `JsonUtils.flatten()` logic changes, thus the storage format is different

Reply:
Don't fully get it. Is my understanding right that V1 is a special case of V2, and therefore the handling logic is subset of V2's?

Reply:
No, v1 and v2 have different format, and the handling logic is also slightly different (notice the delimiter for array access). That's why we need to maintain the version info so that the new query can work on both v1 and v2 segments